beacon_chain/utils/bls/src/lib.rs

```diff
@@ -8,18 +8,31 @@ pub use self::bls_aggregates::PublicKey;
 pub use self::bls_aggregates::SecretKey;
 pub use self::bls_aggregates::Signature;
 
+use std::iter;
+
 pub const BLS_AGG_SIG_BYTE_SIZE: usize = 97;
 
-use hashing::proof_of_possession_hash;
+use hashing::canonical_hash;
+
+fn extend_if_needed(hash: &mut Vec<u8>) {
+    // NOTE: bls_aggregates crate demands 48 bytes, this may be removed as we get closer to production
+    let hash_len = hash.len();
+    if hash_len < 48 {
```

Comment:
Do you think `Vec::resize` would be more suitable here?

https://doc.rust-lang.org/std/vec/struct.Vec.html#method.resize

Reply:
i, in fact, do :)

i'll update now -- thanks for suggesting